cli/src/main.rs

```diff
@@ -28,52 +31,87 @@ struct Args {
+    if !quiet {
+        println!("{}", toml::to_string_pretty(&settings).unwrap());
+    }
 
-    let glob_pattern = if is_dir {
-        format!("{}/**/*.rs", &args.input_pattern)
+    if args.stdin.unwrap() {
+        match format_stdin_result(settings) {
+            Ok(_) => {}
+            Err(err) => eprintln!("{}", err),
+        };
     } else {
-        args.input_pattern
-    };
-
-    let file_paths: Vec<_> = glob(&glob_pattern)
-        .expect("failed to read glob pattern")
-        .collect();
-
-    let total_files = file_paths.len();
-    let start_formatting = Instant::now();
-    file_paths.into_par_iter().for_each(|result| {
-        let print_err = |path: &Path, err| {
-            println!("❌ {}", path.display());
-            eprintln!("\t\t{}", err);
+        let input_pattern = args.input_pattern.unwrap();
+        let is_dir = fs::metadata(&input_pattern)
+            .map(|meta| meta.is_dir())
+            .unwrap_or(false);
+
+        let glob_pattern = if is_dir {
+            format!("{}/**/*.rs", &input_pattern)
+        } else {
+            input_pattern
         };
 
-        match result {
-            Ok(path) => match format_glob_result(&path, settings) {
-                Ok(_) => println!("✅ {}", path.display()),
-                Err(err) => print_err(&path, &err.to_string()),
-            },
-            Err(err) => print_err(err.path(), &err.error().to_string()),
-        };
-    });
-    let end_formatting = Instant::now();
-    println!(
-        "Formatted {} files in {} ms",
-        total_files,
-        (end_formatting - start_formatting).as_millis()
-    )
+        let file_paths: Vec<_> = glob(&glob_pattern)
+            .expect("failed to read glob pattern")
+            .collect();
+
+        let total_files = file_paths.len();
+        let start_formatting = Instant::now();
+        file_paths.into_par_iter().for_each(|result| {
+            let print_err = |path: &Path, err| {
+                println!("❌ {}", path.display());
+                eprintln!("\t\t{}", err);
+            };
+
+            match result {
+                Ok(path) => match format_glob_result(&path, settings) {
+                    Ok(_) => {
+                        if !quiet {
+                            println!("✅ {}", path.display())
+                        }
+                    }
+                    Err(err) => print_err(&path, &err.to_string()),
+                },
+                Err(err) => print_err(err.path(), &err.error().to_string()),
+            };
+        });
+        let end_formatting = Instant::now();
+        if !quiet {
+            println!(
+                "Formatted {} files in {} ms",
+                total_files,
+                (end_formatting - start_formatting).as_millis()
+            )
+        }
+    }
+}
+
+fn format_stdin_result(settings: FormatterSettings) -> anyhow::Result<()> {
+    let buf = &mut Vec::new();
+    let _ = std::io::stdin().read_to_end(buf)?;
```

Comment:
you can use `read_to_string` here: https://doc.rust-lang.org/stable/std/io/trait.Read.html#method.read_to_string